src/main/scala/nl/knaw/dans/easy/fedoratobag/FileInfo.scala

```diff
@@ -91,8 +84,9 @@ object FileInfo extends DebugEnhancedLogging {
           fileMetadata <- FoXml.getFileMD(foXml)
           derivedFromId = derivedFrom(FoXml.getStreamRoot("RELS-EXT", foXml))
           digest = digestValue(FoXml.getStreamRoot("EASY_FILE", foXml))
+          filename = (fileMetadata \\ "name").map(_.text).mkString
           path = (fileMetadata \\ "path").map(_.text).headOption
-            .map(p => Paths.get(replaceNonAllowedCharacters(p)))
+            .map(p => Paths.get(replaceNonAllowedCharactersInPath(p.substring(0, p.lastIndexOf(filename)))+replaceNonAllowedCharactersInFileName(filename)))
```

Comment:
here i remove the filename from the path, because for the path and name different rules apply which chars are allowed. 
However, when the name is not part of the path, or the name does not exist, it fails. how to make this more robust?

Reply:
For full flexibility we can write more verbose unit test spelling out the full XML rather than using the template method as on https://github.com/DANS-KNAW/easy-fedora-to-bag/blob/24307e0854aa794bdb5b350556da5551e726cdea/src/test/scala/nl/knaw/dans/easy/fedoratobag/FileInfosSpec.scala#L72-L75

Reply:
I see it fails without a name at https://github.com/DANS-KNAW/easy-fedora-to-bag/blob/24307e0854aa794bdb5b350556da5551e726cdea/src/main/scala/nl/knaw/dans/easy/fedoratobag/FileInfo.scala#L101